doc/compiler/optimizer/GlobalOptimizationsSummary.md

```diff
@@ -0,0 +1,138 @@
+Basically it scans forward in the basic block from the point of the allocation and
+tries to find explicit definitions of fields (say in constructors) that occur before
+any use of the field. In these cases the zero initialization of fields (required according
+to Java spec, so no field is uninitialized) can be skipped. Note that for reference
+fields the presence of a GC point before any user-specified definition means that
+the field must be zero initialized. Peeking inside calls is done to maximize the
+number of explicit field initializations seen by the analysis. In addition, for
+some special array allocations (e.g. char arrays that are used by `String`, often do
+not need zero initialization as they are filled in by `arraycopy` calls immediately),
+we can skip zero initialization completely.
+
+## Global Copy Propagation (Dead Store Elimination, Dead Structure Removal)
+These three analyses use use/def info to perform the traditional optimizations. The
+analysis is only done for locals (statics/fields omitted). Dead structure removal
+basically tries to find a region of code with no side-effects and having no definition
+that is used outside the structure; in such cases the (dead) region of code can be
+eliminated.
+
+## Redundant Monitor Elimination
+Redundant Monitor Elimination uses value numbering to perform nested monitor removal
+(e.g. syncenter on the same object twice followed by two syncexits). Use value
+number information and structure to perform lock coarsening (e.g. syncenter/syncexit
+on some object followed by another syncenter/syncexit on the same object). In this
+case the extent of code across which the lock is held is increased and one syncexit/syncenter
+can be removed. Deadlock avoidance is ensured by examining the code where the lock
+would be held after coarsening and making sure there are no calls or lock/unlock
+operations in that region of code. Sometimes additional syncenters/syncexits may be
+added on ‘cold’ paths to avoid some syncenters/syncexits on ‘hot’ paths.
+
+## Global Register Allocator (GRA)
+GRA is a tactical solution for achieving the effect of global register assignment,
+even though the underlying code generators have local register allocators. Basic idea
+is that global register dependencies are introduced in the IL trees; these dependencies
+(when evaluated by the code generators) ensure that a certain value is in a certain
+register on all control flow paths into a basic block. Stores/loads from memory are
+changed into stores/loads from registers in the IL (register numbers specify which
+physical register the value is placed into). Global register candidates are chosen
+by various optimizer passes like PRE, induction variable detection etc. and appropriate
+priorities are given to the candidates based on the number of references and live
+range (number of blocks). Advantages are that most of the code is platform neutral
```

Comment:
```suggestion
range (number of blocks). The advantages are that most of the code is platform neutral
```

Reply:
Suggestion : `most of the code is platform neutral and is` -> `most of the code is platform neutral apart from a few simple evaluators that must be added to a new platform code generator, thus drastically reducing the development cost of adding a powerful global register allocator. It is also `